src/journal/journald-syslog.c

```diff
@@ -328,29 +333,37 @@ void server_process_syslog_message(
         struct iovec iovec[N_IOVEC_META_FIELDS + 6];
         unsigned n = 0;
         int priority = LOG_USER | LOG_INFO;
-        _cleanup_free_ char *identifier = NULL, *pid = NULL;
+        _cleanup_free_ char *identifier = NULL, *pid = NULL, *m = NULL;
+        char *syslog_message;
         const char *orig;
 
         assert(s);
         assert(buf);
 
         orig = buf;
+
         syslog_parse_priority(&buf, &priority, true);
 
         if (s->forward_to_syslog)
-                forward_syslog_raw(s, priority, orig, ucred, tv);
+                forward_syslog_raw(s, priority, orig, buf_len, ucred, tv);
+
+        syslog_message = m = memdup(buf, buf_len);
```

Comment:
yeah, please use stack allocation here.. strndupa(), memdupa() or something like that....

We have anumber of malloc()s already in hotpath here, but this is easy to avoid and we don't want more... we'd rather want less..